src/rules/selector-class-pattern/index.js

```diff
@@ -13,23 +14,46 @@ export const messages = ruleMessages(ruleName, {
   expected: selectorValue => `Expected class selector ".${selectorValue}" to match specified pattern`,
 })
 
-export default function (pattern) {
+export default function (pattern, options) {
   return (root, result) => {
     const validOptions = validateOptions(result, ruleName, {
       actual: pattern,
       possible: [ isRegExp, isString ],
+    }, {
+      actual: options,
+      possible: {
+        resolveNestedSelectors: isBoolean,
+      },
+      optional: true,
     })
     if (!validOptions) { return }
 
+    // In context of a nested selector, some combinations
+    // are tested mutliple times, to avoid triggering multiple
+    // identical errors, we will use an internal list of checked rules
+    var checked = []
```

Comment:
I agree with you, and that is actually, the case, but when you have style like : 

```css
.A {
    .B { }
}
```

The rule will test `.A` twice (once as the parent selector, and once in the full selector `.A .B`

So if a rule like this triggers a warning, it will show the same warning twice. The `checked` array, is here to avoid that.

Your example will work just fine as the `.foo` selector is at different places in the code.